src/main/java/net/mcreator/element/ModElementTypeRegistry.java

```diff
@@ -1,117 +1,130 @@
-		put(ModElementType.PAINTING, new ModTypeRegistration<>(null, 30, PaintingGUI::new, Painting.class));
-		put(ModElementType.PARTICLE, new ModTypeRegistration<>(null, 31, ParticleGUI::new, Particle.class));
-		put(ModElementType.GAMERULE, new ModTypeRegistration<>(null, 32, GameRuleGUI::new, GameRule.class));
-	}};
-
-	public static class ModTypeRegistration<GE extends GeneratableElement> {
-
-		private final int icon_id;
-		private final Character shortcut;
-
-		private final ModElementGUIProvider<GE> modElementGUIProvider;
-		private final Class<? extends GE> modElementStorageClass;
-
-		private boolean hasProcedureTriggers;
-
-		ModTypeRegistration(Character shortcut, int icon_id, ModElementGUIProvider<GE> modElementGUIProvider,
-				Class<? extends GE> modElementStorageClass) {
-			this.icon_id = icon_id;
-			this.modElementGUIProvider = modElementGUIProvider;
-			this.shortcut = shortcut;
-			this.modElementStorageClass = modElementStorageClass;
-
-			for (Field field : modElementStorageClass.getFields())
-				if (field.getType().isAssignableFrom(net.mcreator.element.parts.Procedure.class)) {
-					hasProcedureTriggers = true;
-					break;
-				}
-		}
-
-		public Character getShortcut() {
-			return shortcut;
-		}
-
-		public int getIconID() {
-			return icon_id;
-		}
-
-		public ModElementGUI<GE> getModElement(MCreator mcreator, ModElement modElement, boolean editingMode) {
-			return modElementGUIProvider.get(mcreator, modElement, editingMode);
-		}
-
-		public Class<? extends GeneratableElement> getModElementStorageClass() {
-			return modElementStorageClass;
-		}
-
-		public boolean hasProcedureTriggers() {
-			return hasProcedureTriggers;
-		}
-
-		private interface ModElementGUIProvider<GE extends GeneratableElement> {
-			ModElementGUI<GE> get(MCreator mcreator, ModElement modElement, boolean editingMode);
-		}
-
-	}
-
-}
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element;
+
+import net.mcreator.element.types.*;
+import net.mcreator.ui.modgui.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ModElementTypeRegistry {
+
+	public static List<ModElementType<?>> REGISTRY = new ArrayList<>();
+
+	public static void registryModElements() {
```

Comment:
Rename to loadModElements

Reply:
I named it registerModElements, so when we will add support for custom mod elements, we won't have to rename ot again.

Reply:
Even then, it will load them, and all methods for loading things start with load so far